specs/order_spec.rb

```diff
@@ -1,110 +1,293 @@
+      @normal_order.add_product("salad", 4.25)
+      expected_count = @normal_products.size + 1
 
-      order.add_product("salad", 4.25)
-      expected_count = before_count + 1
-      order.products.count.must_equal expected_count
+      @normal_order.products.count.must_equal expected_count # assertion
     end
 
+    # Tests if program adds new product to products.
     it "Is added to the collection of products" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
-      order = Grocery::Order.new(1337, products)
+      @normal_order.add_product("sandwich", 4.25)
 
-      order.add_product("sandwich", 4.25)
-      order.products.include?("sandwich").must_equal true
+      @normal_order.products.include?("sandwich").must_equal true # assertion
+    end
+
+    it "Returns true if the product is new" do
+      @normal_order.add_product("salad", 4.25).must_equal true # assertion
     end
 
     it "Returns false if the product is already present" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_total = @normal_order.total
+      result_of_add = @normal_order.add_product("banana", 4.25) # already has
+      after_total = @normal_order.total
 
-      order = Grocery::Order.new(1337, products)
-      before_total = order.total
+      result_of_add.must_equal false # assertion
+      before_total.must_equal after_total # assertion
+    end
+  end
 
-      result = order.add_product("banana", 4.25)
-      after_total = order.total
+  describe "#remove_product" do
+    # Tests if add_product increases products size.
+    it "Decreases the number of products" do
+      @normal_order.remove_product("banana")
+      expected_count = @normal_products.size - 1
 
-      result.must_equal false
-      before_total.must_equal after_total
+      @normal_order.products.count.must_equal expected_count # assertion
     end
 
-    it "Returns true if the product is new" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
-      order = Grocery::Order.new(1337, products)
+    # Tests if program adds new product to products.
+    it "Is added to the collection of products" do
+      @normal_order.remove_product("banana")
 
-      result = order.add_product("salad", 4.25)
-      result.must_equal true
+      @normal_order.products.include?("banana").must_equal false # assertion
+    end
+
+    it "Returns true if the product is removed" do
+      @normal_order.remove_product("banana").must_equal true # assertion
+    end
+
+    it "Returns false if the product isn't a product" do
+      before_total = @normal_order.total
+      result_of_remove = @normal_order.remove_product("corn") # doesn't have
+      after_total = @normal_order.total
+
+      result_of_remove.must_equal false # assertion
+      before_total.must_equal after_total # assertion
     end
   end
+
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+
+describe "Order Wave 2" do
+
+  expected_first_id = 1
```

Comment:
You should put this in a `before` block as instance variables.  This would keep one test from accidentally changing a variable and then causing an error in another test.